src/schema/basic_classes.yaml

```diff
@@ -454,6 +454,9 @@ slots:
     range: WorkflowExecution
```

Comment:
Based on my **partial** understanding of what the combination of `any_of` and `range` means to some people, I'm wondering whether `range: WorkflowExecution` is now too specific, since `DataGeneration` (i.e. the value of `any_of[1].range`) is a _sibling_ of `WorkflowExecution`, not a _descendant_ of it. 🤷

Reply:
Thanks @eecavanna questions like these are helpful and it seems like @sierra-moxon has the most fully formed understanding of the implementation in LinkML.

Ultimately, I would want to check your proposal (and really all of the permutations of direct `range`s and `any_of.range`s with tests.

Reply:
But in the mean time, I am inclined to say that we should have a consistent style convention: when there are `any_of.range`s, what should the direct `range` assertion be? The closest common parent? LinkML's "Any" type? (I hope not... that doesn't seem ready for prime time to me)?

Reply:
To clarify this PR matches the style of what is in nmdc-schema main. https://microbiomedata.github.io/nmdc-schema/was_generated_by/. The PR that led to that is [here](https://github.com/microbiomedata/nmdc-schema/pull/1978/files#diff-4e6a83888b253648ab6aecbc9754fca181a3826ceb0a1c4f52926e34f69d842b), authored by @turbomam. If there is different guidance and/or that format is causing trouble it should be updated in both places.